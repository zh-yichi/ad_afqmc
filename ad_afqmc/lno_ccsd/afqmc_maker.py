from jax import numpy as jnp
from jax import vmap

def walker_norhf_energy(walker,ham_data,trial_mo):
    '''rhf walker's energy with non-orthonal trial'''
    rot_h1,rot_chol = ham_data['rot_h1'],ham_data['rot_chol']
    nocc = rot_h1.shape[0]
    green_walker = (walker.dot(jnp.linalg.inv(trial_mo.T.conj() @ walker))).T
    f = jnp.einsum('gia,ak->gik',rot_chol[:,:nocc,nocc:],
                   green_walker.T[nocc:,:nocc],optimize='optimal')
    tf = vmap(jnp.trace)(f)
    tc = vmap(jnp.trace)(rot_chol[:,:nocc,:nocc])
    ene11 = 2*jnp.einsum('ia,ia->',green_walker[:nocc,nocc:],rot_h1[:nocc,nocc:])
    ene12 = 2*jnp.sum(tf*tc) \
            - jnp.einsum('gik,gki->',f,rot_chol[:,:nocc,:nocc])
    ene1 = ene11-2*ene12
    ene2 = 2*jnp.sum(tf*tf) \
           - jnp.einsum('gij,gji->',f,f)
    return ene1+ene2 #,ene1,ene2,

def walker_norhf_orb_en(walker,ham_data,wave_data):
    '''rhf walker's energy with non-orthonal trial'''
    rot_h1,rot_chol = ham_data['rot_h1'],ham_data['rot_chol']
    m = jnp.dot(wave_data["prjlo"].T,wave_data["prjlo"])
    nocc = rot_h1.shape[0]
    green_walker = (walker.dot(jnp.linalg.inv(
        wave_data['mo_coeff'].T.conj() @ walker))).T
    f = jnp.einsum('gia,ak->gik',rot_chol[:,:nocc,nocc:],
                   green_walker.T[nocc:,:nocc],optimize='optimal')
    tf = vmap(jnp.trace)(f)
    # tc = vmap(jnp.trace)(rot_chol[:,:nocc,:nocc])
    ene11 = 2*jnp.einsum('ia,ka,ik->',green_walker[:nocc,nocc:],rot_h1[:nocc,nocc:],m)
    ene12 = 2*jnp.einsum('gik,ik,gjj->',f,m,rot_chol[:,:nocc,:nocc]) \
            - jnp.einsum('gij,gjk,ik->',f,rot_chol[:,:nocc,:nocc],m)
    ene1 = ene11-2*ene12
    ene2 = 2*jnp.einsum('gik,ik,g->',f,m,tf) \
           - jnp.einsum('gij,gjk,ik->',f,f,m)
    return ene1+ene2