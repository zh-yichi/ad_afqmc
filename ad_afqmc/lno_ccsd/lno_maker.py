import numpy as np

def thouless_trans(t1):
    q, r = np.linalg.qr(t1)
    u_ai = r.T
    u_ji = q
    u_occ = np.vstack((u_ji,u_ai))
    u, _, _ = np.linalg.svd(u_occ)
    return u
