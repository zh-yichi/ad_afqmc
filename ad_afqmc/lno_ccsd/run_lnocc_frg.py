from functools import partial
from jax import random
#from mpi4py import MPI
import numpy as np
from jax import numpy as jnp
from ad_afqmc import config, wavefunctions, stat_utils
from ad_afqmc.lno_ccsd import lno_ccsd
import time

print = partial(print, flush=True)

ham_data, ham, prop, trial, wave_data, sampler, observable, options, _ = (
    lno_ccsd.prep_lnoccsd_afqmc())

if options["use_gpu"]:
    config.afqmc_config["use_gpu"] = True

config.setup_jax()
MPI = config.setup_comm()

# init_time = time.time()
comm = MPI.COMM_WORLD
rank = comm.Get_rank()  # Process rank
size = comm.Get_size()  # Total number of processes


### initialize propagation
seed = options["seed"]
propagator = prop
init_walkers = None
ham_data = wavefunctions.rhf(trial.norb, trial.nelec,n_batch=trial.n_batch
                                )._build_measurement_intermediates(ham_data, wave_data)
ham_data = ham.build_measurement_intermediates(ham_data, trial, wave_data)
ham_data = ham.build_propagation_intermediates(ham_data, propagator, trial, wave_data)

prop_data = propagator.init_prop_data(trial, wave_data, ham_data, init_walkers)
if jnp.abs(jnp.sum(prop_data["overlaps"])) < 1.0e-6:
    raise ValueError(
        "Initial overlaps are zero. Pass walkers with non-zero overlap."
    )
prop_data["key"] = random.PRNGKey(seed + rank)

prop_data["overlaps"] = trial.calc_overlap(prop_data["walkers"], wave_data)
prop_data["n_killed_walkers"] = 0
prop_data["pop_control_ene_shift"] = prop_data["e_estimate"]

### relaxation ###
init_time = time.time()
comm.Barrier()
if rank == 0:
    hf_orb_cr,hf_orb_en = lno_ccsd._frg_hf_cr(
        ham_data['rot_h1'], ham_data['rot_chol'],prop_data["walkers"][0],trial,wave_data)
    cc_orb_cr = lno_ccsd._frg_ccsd_cr(
        prop_data["walkers"][0],ham_data,wave_data,trial,1e-6)
    
    cc_orb_en = hf_orb_cr+cc_orb_cr
    e_init = prop_data["e_estimate"]

    print(f'# afqmc propagation with {options["n_walkers"]*size} walkers')
    print(f'# appraching equilibrium')
    print('# step  energy  hf_orb_en  cc_orb_en  time')
    print(f"  {0:3d}"
          f"  {e_init:.6f}"
          f"  {hf_orb_en:.6f}"
          f"  {cc_orb_en:.6f}"
          f"  {time.time() - init_time:.2f} "
        )
comm.Barrier()

for n in range(options["n_eql"]):
    prop_data, (blk_en,blk_wt,blk_hf_orb_en,blk_cc_orb_en) \
                = lno_ccsd.propagate_phaseless_orb(
                    ham_data,prop,prop_data,trial,wave_data,sampler)
    
    blk_en = np.array([blk_en], dtype="float32")
    blk_wt = np.array([blk_wt], dtype="float32")
    blk_hf_orb_en = np.array([blk_hf_orb_en], dtype="float32")
    blk_cc_orb_en = np.array([blk_cc_orb_en], dtype="float32")    

    blk_wt_en = np.array(
        [blk_en * blk_wt], dtype="float32"
    )
    blk_wt_hf_orb_en = np.array(
        [blk_hf_orb_en * blk_wt], dtype="float32"
    )
    blk_wt_cc_orb_en = np.array(
        [blk_cc_orb_en * blk_wt], dtype="float32"
    )

    tot_wt_en = np.zeros(1, dtype="float32")
    tot_wt = np.zeros(1, dtype="float32")
    tot_wt_hf_orb_en = np.zeros(1, dtype="float32")
    tot_wt_cc_orb_en = np.zeros(1, dtype="float32")

    comm.Reduce(
            [blk_wt_en, MPI.FLOAT],
            [tot_wt_en, MPI.FLOAT],
            op=MPI.SUM,
            root=0,
        )
    comm.Reduce(
        [blk_wt, MPI.FLOAT],
        [tot_wt, MPI.FLOAT],
        op=MPI.SUM,
        root=0,
    )
    comm.Reduce(
            [blk_wt_hf_orb_en, MPI.FLOAT],
            [tot_wt_hf_orb_en, MPI.FLOAT],
            op=MPI.SUM,
            root=0,
        )
    comm.Reduce(
            [blk_wt_cc_orb_en, MPI.FLOAT],
            [tot_wt_cc_orb_en, MPI.FLOAT],
            op=MPI.SUM,
            root=0,
        )

    comm.Barrier()
    if rank == 0:
        blk_en = tot_wt_en / tot_wt
        blk_wt = tot_wt
        blk_hf_orb_en = tot_wt_hf_orb_en / tot_wt
        blk_cc_orb_en = tot_wt_cc_orb_en / tot_wt

    comm.Bcast(blk_en, root=0)
    comm.Bcast(blk_wt, root=0)
    comm.Bcast(blk_hf_orb_en, root=0)
    comm.Bcast(blk_cc_orb_en, root=0)
    
    prop_data = propagator.orthonormalize_walkers(prop_data)
    prop_data = propagator.stochastic_reconfiguration_global(prop_data, comm)
    prop_data["e_estimate"] = (
         0.9 * prop_data["e_estimate"] + 0.1 * blk_en[0]
         )
    comm.Barrier()

    comm.Barrier()
    if rank == 0:
        print(
            f"  {n+1:3d}"
            f"  {blk_en[0]:.6f}"
            f"  {blk_hf_orb_en[0]:.6f}"
            f"  {blk_cc_orb_en[0]:.6f}"
            f"  {time.time() - init_time:.2f} "
        )
    comm.Barrier()

glb_blk_en = None
glb_blk_wt = None
glb_blk_hf_orb_en = None
glb_blk_cc_orb_en = None

if rank == 0:
    glb_blk_en = np.zeros(size * sampler.n_blocks)
    glb_blk_wt = np.zeros(size * sampler.n_blocks)
    glb_blk_hf_orb_en = np.zeros(size * sampler.n_blocks)
    glb_blk_cc_orb_en = np.zeros(size * sampler.n_blocks)
    
comm.Barrier()
if rank == 0:
    print("# Sampling sweeps:")
    print("# iter  energy  err  hf_orb_en  err  cc_orb_cr  err  time")
comm.Barrier()

for n in range(sampler.n_blocks):
    prop_data,(blk_en,blk_wt,blk_hf_orb_en,blk_cc_orb_en) \
        = lno_ccsd.propagate_phaseless_orb(
            ham_data,prop,prop_data,trial,wave_data,sampler)

    blk_en = np.array([blk_en], dtype="float32")
    blk_wt = np.array([blk_wt], dtype="float32")
    blk_hf_orb_en = np.array([blk_hf_orb_en], dtype="float32")
    blk_cc_orb_en = np.array([blk_cc_orb_en], dtype="float32")

    gather_en = None
    gather_wt = None
    gather_hf_orb_en = None
    gather_cc_orb_en = None

    comm.Barrier()
    if rank == 0:
        gather_en = np.zeros(size, dtype="float32")
        gather_wt = np.zeros(size, dtype="float32")
        gather_hf_orb_en = np.zeros(size, dtype="float32")
        gather_cc_orb_en = np.zeros(size, dtype="float32")
    comm.Barrier()

    comm.Gather(blk_en, gather_en, root=0)
    comm.Gather(blk_wt, gather_wt, root=0)
    comm.Gather(blk_hf_orb_en, gather_hf_orb_en, root=0)
    comm.Gather(blk_cc_orb_en, gather_cc_orb_en, root=0)

    comm.Barrier()
    if rank == 0:
        glb_blk_en[n * size : (n + 1) * size] = gather_en
        glb_blk_wt[n * size : (n + 1) * size] = gather_wt
        glb_blk_hf_orb_en[n * size : (n + 1) * size] = gather_hf_orb_en
        glb_blk_cc_orb_en[n * size : (n + 1) * size] = gather_cc_orb_en

        assert gather_wt is not None

        blk_wt= np.sum(gather_wt)
        blk_en = np.sum(gather_wt * gather_en) / blk_wt
    comm.Barrier()
    
    prop_data = propagator.orthonormalize_walkers(prop_data)
    prop_data = propagator.stochastic_reconfiguration_global(prop_data, comm)
    prop_data["e_estimate"] = 0.9 * prop_data["e_estimate"] + 0.1 * blk_en

    if n % (max(sampler.n_blocks // 10, 1)) == 0:
        comm.Barrier()
        if rank == 0:
                en, en_err = stat_utils.blocking_analysis(
                    glb_blk_wt[: (n + 1) * size],
                    glb_blk_en[: (n + 1) * size],
                    neql=0,
                )
                hf_orb_en, hf_orb_en_err = stat_utils.blocking_analysis(
                    glb_blk_wt[: (n + 1) * size],
                    glb_blk_hf_orb_en[: (n + 1) * size],
                    neql=0,
                )
                cc_orb_en, cc_orb_en_err = stat_utils.blocking_analysis(
                    glb_blk_wt[: (n + 1) * size],
                    glb_blk_cc_orb_en[: (n + 1) * size],
                    neql=0,
                )

                if en_err is not None:
                    en_err = f"{en_err:.6f}"
                else:
                    en_err = f"  {en_err}  "
                if hf_orb_en_err is not None:
                    hf_orb_en_err = f"{hf_orb_en_err:.6f}"
                else:
                    hf_orb_en_err = f"  {hf_orb_en_err}  "
                if cc_orb_en_err is not None:
                    cc_orb_en_err = f"{cc_orb_en_err:.6f}"
                else:
                    cc_orb_en_err = f"  {cc_orb_en_err}  "

                en = f"{en:.6f}"
                hf_orb_en = f"{hf_orb_en:.6f}"
                cc_orb_en = f"{cc_orb_en:.6f}"

                print(f"  {n:4d}  {en}  {en_err}  {hf_orb_en}  {hf_orb_en_err}"
                      f"  {cc_orb_en}  {cc_orb_en_err}  {time.time() - init_time:.2f}")
        comm.Barrier()


comm.Barrier()
if rank == 0:
    assert glb_blk_en is not None
    assert glb_blk_wt is not None
    assert glb_blk_hf_orb_en is not None
    assert glb_blk_cc_orb_en is not None


    samples_clean, idx = stat_utils.reject_outliers(
    np.stack(
                (
                    glb_blk_wt,
                    glb_blk_en,
                    glb_blk_hf_orb_en,
                    glb_blk_cc_orb_en,
                )
            ).T,
            1,
        )

    print(
        f"# Number of outliers in post: {glb_blk_wt.size - samples_clean.shape[0]} "
        )
    
    glb_blk_wt = samples_clean[:, 0]
    glb_blk_en = samples_clean[:, 1]
    glb_blk_hf_orb_en = samples_clean[:, 2]
    glb_blk_cc_orb_en = samples_clean[:, 3]

    en, en_err = stat_utils.blocking_analysis(
                    glb_blk_wt[: (n + 1) * size],
                    glb_blk_en[: (n + 1) * size],
                    neql=0,printQ=True
                )
    hf_orb_en, hf_orb_en_err = stat_utils.blocking_analysis(
                    glb_blk_wt[: (n + 1) * size],
                    glb_blk_hf_orb_en[: (n + 1) * size],
                    neql=0,printQ=True
                )
    cc_orb_en, cc_orb_en_err = stat_utils.blocking_analysis(
                    glb_blk_wt[: (n + 1) * size],
                    glb_blk_cc_orb_en[: (n + 1) * size],
                    neql=0,printQ=True
                )
    
    if en_err is not None:
        en_err = f"{en_err:.6f}"
    else:
        en_err = f"  {en_err}  "
    if hf_orb_en_err is not None:
        hf_orb_en_err = f"{hf_orb_en_err:.6f}"
    else:
        hf_orb_en_err = f"  {hf_orb_en_err}  "
    if cc_orb_en_err is not None:
        cc_orb_en_err = f"{cc_orb_en_err:.6f}"
    else:
        cc_orb_en_err = f"  {cc_orb_en_err}  "


    en = f"{en:.6f}"
    hf_orb_en = f"{hf_orb_en:.6f}"
    cc_orb_en = f"{cc_orb_en:.6f}"

    print(f"Final Results:")
    print(f"lno-ccsd-afqmc energy: {en} +/- {en_err}")
    print(f"lno-ccsd-afqmc hf_orb_en: {hf_orb_en} +/- {hf_orb_en_err}")
    print(f"lno-ccsd-afqmc cc_orb_en: {cc_orb_en} +/- {cc_orb_en_err}")
    print(f"total run time: {time.time() - init_time:.2f}")

comm.Barrier()
