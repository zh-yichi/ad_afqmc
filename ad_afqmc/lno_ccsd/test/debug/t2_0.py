from pyscf import gto, scf, cc
from ad_afqmc.lno_ccsd import lno_ccsd, lno_maker
import os
# import numpy as np

# T shape H2 dimer
# d = 3
# atoms = f'''
# H     0.000     0.000    -0.370
# H     0.000     0.000     0.370
# H     {d}       0.000     0.000
# H     {0.74+d}  0.000     0.000
# '''

a = 1. # 2aB
nH = 6
atoms = ""
for i in range(nH):
    atoms += f"H {i*a:.5f} 0.00000 0.00000 \n"

mol = gto.M(atom=atoms, basis="sto6g", verbose=4)
mf = scf.RHF(mol)
mf.kernel()

mycc = cc.CCSD(mf)
mycc.kernel()

options = {'n_eql': 5,
           'n_prop_steps': 30,
            'n_ene_blocks': 1,
            'n_sr_blocks': 10,
            'n_blocks': 5,
            'n_walkers': 1,
            'seed': 98,
            'walker_type': 'rhf',
            'trial': 'cisd',
            'dt':0.005,
            'ad_mode':None,
            'use_gpu': False,
            }

# u = lno_maker.thouless_trans(mycc.t1)
# mo_t = mf.mo_coeff @ u
# mf.mo_coeff = mo_t

import numpy as np
t2 = mycc.t2
t2_0 = np.zeros(t2.shape)
mycc.t2 = t2_0
from ad_afqmc import pyscf_interface, run_afqmc
pyscf_interface.prep_afqmc(mycc,chol_cut=1e-7)

from mpi4py import MPI
MPI.Finalize()
run_afqmc.run_afqmc(options=options, nproc=5)