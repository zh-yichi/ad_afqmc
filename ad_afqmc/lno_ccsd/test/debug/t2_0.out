#INFO: **** input file is /home/yichi/research/software/cs_afqmc/ad_afqmc/lno_ccsd/test/debug/t2_0.py ****
from pyscf import gto, scf, cc
from ad_afqmc.lno_ccsd import lno_ccsd, lno_maker
import os
# import numpy as np

# T shape H2 dimer
# d = 3
# atoms = f'''
# H     0.000     0.000    -0.370
# H     0.000     0.000     0.370
# H     {d}       0.000     0.000
# H     {0.74+d}  0.000     0.000
# '''

a = 1. # 2aB
nH = 6
atoms = ""
for i in range(nH):
    atoms += f"H {i*a:.5f} 0.00000 0.00000 \n"

mol = gto.M(atom=atoms, basis="sto6g", verbose=4)
mf = scf.RHF(mol)
mf.kernel()

mycc = cc.CCSD(mf)
mycc.kernel()

options = {'n_eql': 5,
           'n_prop_steps': 30,
            'n_ene_blocks': 1,
            'n_sr_blocks': 10,
            'n_blocks': 5,
            'n_walkers': 1,
            'seed': 98,
            'walker_type': 'rhf',
            'trial': 'cisd',
            'dt':0.005,
            'ad_mode':None,
            'use_gpu': False,
            }

# u = lno_maker.thouless_trans(mycc.t1)
# mo_t = mf.mo_coeff @ u
# mf.mo_coeff = mo_t

import numpy as np
t2 = mycc.t2
t2_0 = np.zeros(t2.shape)
mycc.t2 = t2_0
from ad_afqmc import pyscf_interface, run_afqmc
pyscf_interface.prep_afqmc(mycc,chol_cut=1e-7)

from mpi4py import MPI
MPI.Finalize()
run_afqmc.run_afqmc(options=options, nproc=5)#INFO: ******************** input file end ********************


System: uname_result(system='Linux', node='yichi-thinkpad', release='4.4.0-26100-Microsoft', version='#5074-Microsoft Fri Jan 01 08:00:00 PST 2016', machine='x86_64')  Threads 12
Python 3.10.16 | packaged by conda-forge | (main, Dec  5 2024, 14:16:10) [GCC 13.3.0]
numpy 1.24.3  scipy 1.14.1  h5py 3.12.1
Date: Tue Sep 23 17:07:47 2025
PySCF version 2.8.0
PySCF path  /home/yichi/research/software/lno_pyscf
GIT HEAD (branch master) ef75f4190e4de208685670651dc6c467f72b6794

[ENV] PYSCF_EXT_PATH /home/yichi/research/software/pyscf
[CONFIG] conf_file None
[INPUT] verbose = 4
[INPUT] num. atoms = 6
[INPUT] num. electrons = 6
[INPUT] charge = 0
[INPUT] spin (= nelec alpha-beta = 2S) = 0
[INPUT] symmetry False subgroup None
[INPUT] Mole.unit = angstrom
[INPUT] Symbol           X                Y                Z      unit          X                Y                Z       unit  Magmom
[INPUT]  1 H      0.000000000000   0.000000000000   0.000000000000 AA    0.000000000000   0.000000000000   0.000000000000 Bohr   0.0
[INPUT]  2 H      1.000000000000   0.000000000000   0.000000000000 AA    1.889726124565   0.000000000000   0.000000000000 Bohr   0.0
[INPUT]  3 H      2.000000000000   0.000000000000   0.000000000000 AA    3.779452249130   0.000000000000   0.000000000000 Bohr   0.0
[INPUT]  4 H      3.000000000000   0.000000000000   0.000000000000 AA    5.669178373695   0.000000000000   0.000000000000 Bohr   0.0
[INPUT]  5 H      4.000000000000   0.000000000000   0.000000000000 AA    7.558904498260   0.000000000000   0.000000000000 Bohr   0.0
[INPUT]  6 H      5.000000000000   0.000000000000   0.000000000000 AA    9.448630622825   0.000000000000   0.000000000000 Bohr   0.0

nuclear repulsion = 4.603841735004
number of shells = 6
number of NR pGTOs = 36
number of NR cGTOs = 6
basis = sto6g
ecp = {}
CPU time:         1.19


******** <class 'pyscf.scf.hf.RHF'> ********
method = RHF
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
diis_damp = 0
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /tmp/tmpl9r8md67
max_memory 4000 MB (current use 136 MB)
Set gradient conv threshold to 3.16228e-05
Initial guess from minao.
init E= -2.43456844968186
  HOMO = -0.206186844053804  LUMO = 0.134147801579113
cycle= 1 E= -3.1487806963508  delta_E= -0.714  |g|= 0.0984  |ddm|= 1.71
  HOMO = -0.322364050884791  LUMO = 0.205265687081062
cycle= 2 E= -3.15516555755206  delta_E= -0.00638  |g|= 0.0314  |ddm|= 0.222
  HOMO = -0.32657231270121  LUMO = 0.219132025413817
cycle= 3 E= -3.15597595168783  delta_E= -0.00081  |g|= 0.00575  |ddm|= 0.0797
  HOMO = -0.329504962281801  LUMO = 0.219324890545479
cycle= 4 E= -3.15600091849169  delta_E= -2.5e-05  |g|= 0.000147  |ddm|= 0.018
  HOMO = -0.329429609111351  LUMO = 0.219283193175082
cycle= 5 E= -3.15600092949307  delta_E= -1.1e-08  |g|= 8.09e-06  |ddm|= 0.000293
  HOMO = -0.329440909371494  LUMO = 0.21928831661454
cycle= 6 E= -3.1560009295473  delta_E= -5.42e-11  |g|= 1.06e-07  |ddm|= 2.49e-05
  HOMO = -0.329441053062924  LUMO = 0.219288365072638
Extra cycle  E= -3.15600092954731  delta_E= -9.77e-15  |g|= 3.42e-08  |ddm|= 1.93e-07
converged SCF energy = -3.15600092954731

******** <class 'pyscf.cc.ccsd.CCSD'> ********
CC2 = 0
CCSD nocc = 3, nmo = 6
max_cycle = 50
direct = 0
conv_tol = 1e-07
conv_tol_normt = 1e-05
diis_space = 6
diis_start_cycle = 0
diis_start_energy_diff = 1e+09
max_memory 4000 MB (current use 142 MB)
Init t2, MP2 energy = -3.21911395512482  E_corr(MP2) -0.0631130255775121
Init E_corr(CCSD) = -0.0631130255775127
cycle = 1  E_corr(CCSD) = -0.0845186863080102  dE = -0.0214056607  norm(t1,t2) = 0.0635371
cycle = 2  E_corr(CCSD) = -0.093478483708098  dE = -0.0089597974  norm(t1,t2) = 0.0329352
cycle = 3  E_corr(CCSD) = -0.101050325428887  dE = -0.00757184172  norm(t1,t2) = 0.0171821
cycle = 4  E_corr(CCSD) = -0.101343622036368  dE = -0.000293296607  norm(t1,t2) = 0.00397072
cycle = 5  E_corr(CCSD) = -0.101172438387178  dE = 0.000171183649  norm(t1,t2) = 0.00161877
cycle = 6  E_corr(CCSD) = -0.101226441797361  dE = -5.40034102e-05  norm(t1,t2) = 0.000300468
cycle = 7  E_corr(CCSD) = -0.101212334727656  dE = 1.41070697e-05  norm(t1,t2) = 9.39978e-05
cycle = 8  E_corr(CCSD) = -0.101212433529271  dE = -9.88016152e-08  norm(t1,t2) = 2.26032e-05
cycle = 9  E_corr(CCSD) = -0.101213141225318  dE = -7.07696047e-07  norm(t1,t2) = 4.33698e-06
cycle = 10  E_corr(CCSD) = -0.101213574853364  dE = -4.33628046e-07  norm(t1,t2) = 1.43854e-06
cycle = 11  E_corr(CCSD) = -0.101213601201206  dE = -2.63478423e-08  norm(t1,t2) = 1.75363e-07
CCSD converged
E(CCSD) = -3.257214530748514  E_corr = -0.101213601201206
#
# Preparing AFQMC calculation
# If you import pyscf cc modules and use MPI for AFQMC in the same script, finalize MPI before calling the AFQMC driver.
# Calculating Cholesky integrals
# Finished calculating Cholesky integrals
#
# Size of the correlation space:
# Number of electrons: (3, 3)
# Number of basis functions: 6
# Number of Cholesky vectors: 15
#
# Hostname: yichi-thinkpad
# System Type: Linux
# Machine Type: x86_64
# Processor: x86_64
# Number of MPI ranks: 5
#
# norb: 6
# nelec: (3, 3)
#
# n_eql: 5
# n_prop_steps: 30
# n_ene_blocks: 1
# n_sr_blocks: 10
# n_blocks: 5
# n_walkers: 1
# seed: 98
# walker_type: rhf
# trial: cisd
# dt: 0.005
# use_gpu: False
# n_exp_terms: 6
# orbital_rotation: True
# do_sr: True
# symmetry: False
# save_walkers: False
# ene0: 0.0
# free_projection: False
# n_batch: 1
# LNO: False
# orbE: 0
# maxError: 0.001
#
# Equilibration sweeps:
#   Iter        Block energy      Walltime
#     0      -3.156007484e+00     3.40e+00 
#     1      -3.248546839e+00     1.34e+01 
#     2      -3.237497568e+00     2.22e+01 
#     3      -3.236656427e+00     2.24e+01 
#     4      -3.246036291e+00     2.27e+01 
#     5      -3.256985664e+00     2.29e+01 
#
# Sampling sweeps:
#  Iter        Mean energy          Stochastic error       Walltime
     0      -3.280116510e+00        5.399838862e-02        3.06e+01 
     1      -3.259029642e+00        2.838429788e-02        3.06e+01 
     2      -3.260445125e+00        1.924076309e-02        3.06e+01 
     3      -3.261996309e+00        1.531766352e-02        3.07e+01 
     4      -3.272519770e+00        1.657069590e-02        3.07e+01 
#
# Number of large deviations: 0
# Number of outliers in post: 0 
#
# Mean: -3.27251977e+00
# Block size    # of blocks         Mean                Error
      1               25       -3.27251977e+00       1.657070e-02
      2               12       -3.26387091e+00       9.642367e-03
      3                8       -3.26387091e+00       1.202517e-02
      5                5       -3.27251977e+00       1.400930e-02
     10                2       -3.26199631e+00       4.193279e-03
# Stocahstic error estimate: 1.657070e-02
#
AFQMC energy: -3.27 +/- 0.02

ph_afqmc walltime: 30.879714250564575
